integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/standalone/apiDeploy/AllHttpMethodsForWildcardTestCase.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.choreo.connect.tests.testcases.standalone.apiDeploy;
+
+import com.github.dockerjava.zerodep.shaded.org.apache.hc.core5.http.HttpStatus;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.choreo.connect.mockbackend.dto.EchoResponse;
+import org.wso2.choreo.connect.tests.util.HttpResponse;
+import org.wso2.choreo.connect.tests.util.HttpsClientRequest;
+import org.wso2.choreo.connect.tests.util.TestConstant;
+import org.wso2.choreo.connect.tests.util.TokenUtil;
+import org.wso2.choreo.connect.tests.util.Utils;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+public class AllHttpMethodsForWildcardTestCase {
+    private String jwtTokenProd;
+    private final Map<String, String> headers = new HashMap<>();
+    private static final String API_CONTEXT = "/all_http_methods_for_wildcard";
+
+    @BeforeClass(description = "Get Prod token")
+    void start() throws Exception {
+        jwtTokenProd = TokenUtil.getJwtForPetstore(TestConstant.KEY_TYPE_PRODUCTION, null, false);
+        headers.put(HttpHeaderNames.AUTHORIZATION.toString(), "Bearer " + jwtTokenProd);
+    }
+
+    @Test(description = "test path with trailing slash for GET")
+    public void testTrailingSlashGET() throws Exception {
+        EchoResponse echoResponse = Utils.invokeEchoGet(API_CONTEXT, "/",
```

Comment:
I guess this test is for invoking /api_context/. We need another test case for /api_context. There is no end slash.

Reply:
In APIM if you create an API with /* resource then you can invoke this as /apicontext/ and /apicontext. Can't we achieve the same here?

Reply:
we should fix it then

Reply:
This works. Let me update the tests. Didn't work while writing tests. Must have written the test wrong.

Reply:
fixed with https://github.com/wso2/product-microgateway/pull/2878/commits/10bdb144f96c4a1ffcea84c43ab7803b61415a9a